src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -306,7 +306,10 @@ public void onPlayerEnterNearby(final Player player)
     @Override
     public void onPlacement()
     {
-        ChunkDataHelper.claimBuildingChunks(colony, true, getPosition(), getClaimRadius(getBuildingLevel()), getCorners());
+        if (getBuildingLevel() == 0)
+        {
+            ChunkDataHelper.claimBuildingChunks(colony, true, getPosition(), getClaimRadius(getBuildingLevel()), getCorners());
```

Comment:
Would this properly expand the claim area when a building gets upgraded, or when a building gets pasted in at a higher level than level 0?

Reply:
ye that still triggers upgrades on the building